lib/spack/spack/solver/concretize.lp

```diff
@@ -317,18 +316,21 @@ possible_version_weight(node(ID, Package), Weight)
 
 % If there is at least a version that satisfy the constraint, impose a lower
 % bound on the choice rule to avoid false positives with the error below
-1 { attr("version", node(ID, Package), Version) : pkg_fact(Package, version_satisfies(Constraint, Version)) }
+{ attr("version", node(ID, Package), Version) : pkg_fact(Package, version_satisfies(Constraint, Version)) }
```

Comment:
Pretty sure the comment for this and the rule above is wrong now. What does this do?

I'm not seeing, in version semantics, where we say we can't have two versions assigned to a node. I see it for virtuals but not for regular nodes... did something get lost?

Reply:
This rule is now redundant with the one above, deleted in 78cc553879e338136d4bc2b4499077d4b7ff2785

Reply:
> I'm not seeing, in version semantics, where we say we can't have two versions assigned to a node. I see it for virtuals but not for regular nodes... did something get lost?

This was not changed here. The rules are:
![Screenshot from 2025-03-25 07-48-17](https://github.com/user-attachments/assets/2a6b7410-43ee-4309-94bc-5b27f2f861c0)

Reply:
ah ok awesome -- I missed the generalization of single-value attributes.  That's good.  Should virtuals not also have this? Something for later.